protocol/lib/context.go

```diff
@@ -1,12 +1,35 @@
 package lib
 
 import (
+	"context"
 	"fmt"
+
 	"github.com/cometbft/cometbft/crypto/tmhash"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/lib/log"
 )
 
 type TxHash string
 
 func GetTxHash(tx []byte) TxHash {
 	return TxHash(fmt.Sprintf("%X", tmhash.Sum(tx)))
 }
+
+// UnwrapSDKContext is a thin wrapper around cosmos sdk's unwrap function
+// that extracts the cosmos context from the standard golang context.
+// If moduleName is provided, it appends the persistent log tag with
+// the module name to the logger in the context.
+func UnwrapSDKContext(
```

Comment:
This function is intended to replace the cosmos sdk `UnwrapSDKContext` function. It accepts an additional parameter for module name and any logs generated with the context will have be tagged with module name.